test/appending-linkage-type.ll

```diff
@@ -0,0 +1,33 @@
+; RUN: llvm-as < %s -o %t.bc
+; RUN: llvm-spirv %t.bc -o %t.spv -spirv-ext=+all
```

Comment:
Is `-spirv-ext=+all` really needed?

Reply:
Yes, otherwise **@llvm.compiler.used** disappears during LLVM-IR -> SPIR-V -> LLVM-IR translation. But in general, we only need to make sure that **@llvm.compiler.used** retains **appending** linkage type after SPIR-V -> LLVM-IR translation, so we can replace this **.ll** test by **.spt**. Should we do it this way?

Reply:
> Yes, otherwise @llvm.compiler.used disappears during LLVM-IR -> SPIR-V -> LLVM-IR translation.

Right, in that case you can perhaps explicitly list the extension(s) that this relies on instead of enabling `all`.

> we can replace this .ll test by .spt. Should we do it this way?

I find .ll files easier to understand, so I prefer keeping the .ll (but that's my personal opinion).

Reply:
Done